include/aws/io/socket.h

```diff
@@ -43,6 +50,15 @@ struct aws_socket_options {
      * lost. If zero OS defaults are used. On Windows, this option is meaningless until Windows 10 1703.*/
     uint16_t keep_alive_max_failed_probes;
     bool keepalive;
+
+    /**
+     * (Optional)
+     * This property is used to bind the socket to a particular network interface by name, such as eth0 and ens32.
+     * If this is empty, the socket will not be bound to any interface and will use OS defaults. If the provided name
+     * is invalid, `aws_socket_init()` will error out with AWS_IO_SOCKET_INVALID_OPTIONS. This option is only
+     * supported on Linux and MacOS and will be ignored on other platforms.
```

Comment:
maybe AWS_ERROR_PLATFORM_NOT_SUPPORTED or AWS_IO_SOCKET_INVALID_OPTIONS instead of "ignored on other platforms"

seems better to give an error if someone is trying to use it on windows. Otherwise they'll think it works, but wonder why nothing is different, and spend a bunch of time chasing down what's going wrong

Reply:
I thought about it and thought it's better to keep it cross-platform. Otherwise customers will have to do something like this:
```
#ifdef __linux__
socket_options.interface_name = ...
#endif
```
With the ignoring it approach, they don't need to ifdef.

Reply:
in what world is there a specific name in the user's cross-platform code? Any name is going to be extremely specific

Reply:
Thanks, updated.